### PR TITLE
【generate】add no cache generate

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR Category

 Execute Infrastructure

### PR Types

 Improvements

### Description

支持没有适配kvcache 更新路径的模型，全部使用prefill 完成推理

### 是否引起精度变化

否

### CODE REVIEW

File: src/paddlefleet/generation/greedy_generator.py [hunk 1]
```diff
@@ -324,6 +324,83 @@ def __init__(self, fleet_model: GPTModel):
...
+            for step in range(max_new_tokens):
+                cur_len = generated.shape[1]
+                position_ids = (
+                    paddle.arange(cur_len, dtype="int64")
+                    .unsqueeze(0)
+                    .expand([bsz, cur_len])
+                )
+                prefill_startend = paddle.full(
+                    [bsz, 1, cur_len, 1], cur_len, dtype="int32"
+                )
```
Comment (risemeup1111) [inline src/paddlefleet/generation/greedy_generator.py]: P1 **优先级：P1**

这里在 no_cache 模式下无条件传入 `[B, 1, cur_len, 1]` 的 `attn_mask_startend_row_indices`。对 `gpt_model_use_experimental_version=True` 的模型，`DotProductAttention._ec_compatible_flash_attention()` 只要收到非空 mask 就要求 `shape[-1] == 2`，因此该分支会在首个 prefill 触发 assert；默认 cache 分支已经只在 SWA 需要 flashmask 时才构造这个 1 列 mask。请同步这个条件，非 SWA 场景传 `None`。

**处理要求：请针对该评论修复并提交新的 commit。**

```suggestion
                if self.cache.window_size and any(self.cache.swa_layers):
                    prefill_startend = paddle.full(
                        [bsz, 1, cur_len, 1], cur_len, dtype="int32"
                    )
                else:
                    prefill_startend = None
```

File: src/paddlefleet/generation/greedy_generator.py [hunk 2]
```diff
@@ -324,6 +324,83 @@ def __init__(self, fleet_model: GPTModel):
...
+                if return_log_probs and log_probs_per_batch is not None:
+                    step_log_probs = paddle.nn.functional.log_softmax(
+                        last_logits.cast("float32"), axis=-1
+                    )  # [B, vocab]
+                    for b in range(bsz):
+                        tok_id = int(next_tok[b, 0].item())
+                        log_probs_per_batch[b].append(
+                            float(step_log_probs[b, tok_id].item())
+                        )
```
Comment (risemeup1111) [inline src/paddlefleet/generation/greedy_generator.py]: P1 **优先级：P1**

这里每一步都会给所有 batch 样本追加 log_prob，但 `done` 已经在 no_cache 循环外维护。mixed-EOS batch 中某个样本先命中 EOS、其它样本继续生成时，已完成样本后续 step 的 log_prob 仍会被追加，导致返回长度不再表示该样本实际生成到 EOS 的 token 数；默认 KV cache decode 分支下方已经用 `if not done` 避免了这个问题。请同步该保护，并补一个 no_cache mixed-EOS batch 测试。

**处理要求：请针对该评论修复并提交新的 commit。**

```suggestion
                    for b in range(bsz):
                        if not bool(done[b, 0].item()):
                            tok_id = int(next_tok[b, 0].item())
                            log_probs_per_batch[b].append(
                                float(step_log_probs[b, tok_id].item())
                            )
```

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
复查了新提交，no-cache 采样参数已改为复用 `_sample_next_token`，这一点看起来已解决。仍有两个需要先修复的 no-cache 行为问题，细节已放在行级评论里。当前 CI 还有多项测试在运行；本地目标测试因环境缺少 `paddle` 未能收集。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Review by risemeup1111 — APPROVED** [review]
复查了最新提交，上一轮 no-cache 的两个阻塞问题已经修正，当前未发现新的代码问题。`Build Fleet whl` 的失败看起来是构建镜像拉取超时，属于外部流水线问题。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Review by changeyoung98 — APPROVED** [review]
